File: tests/vb/vb_oop_attributes_events/interface_basic.vb
' vybe-test: vb/vb_oop_attributes_events/interface_basic
' origin: languages/vb/tests/vb/test_vb_oop_attributes_events.rs

' Vybe test harness — Visual Basic.
'
' Real VB source alongside harness/go/check.go and harness/js/check.js, the way
' test262's assert.js is JavaScript.
'
' A test's verdict is its EXIT CODE. __Check prints its diagnostic BEFORE
' throwing: an uncaught exception surfaces as `RuntimeError: [object]`, which
' says nothing at all.

Module VybeCheck
    Sub __Check(got As String, want As String)
        If got <> want Then
            Console.WriteLine("FAIL: want [" & want & "] got [" & got & "]")
            Throw New Exception("assertion failed")
        End If
    End Sub
End Module

Interface I: Sub Test(): End Interface: Class C: Implements I: Public Sub Test() Implements I.Test: __Check(CStr("I"), "I"): End Sub: End Class: Module M: Sub Main(): Dim obj As I = New C(): obj.Test(): End Sub: End Module

File: tests/fortran/types/interface_basic.f90
! vybe-test: fortran/types/interface_basic
! origin: languages/fortran/tests/fortran/test_types.rs

program test
    interface
        function add(a, b) result(res)
            integer, intent(in) :: a, b
            integer :: res
        end function add
    end interface
    if (trim("ok") /= "ok") then
    print *, "FAIL: want [ok] got [", "ok", "]"
    stop 1
end if
end program test
